### PR TITLE
add "don't ask again" option to media warning

### LINKED ISSUE #1915: Give option for user to suppress warning on opening attachments
Friends who are using TextSecure have found it frustrating that it alerts them every time they want to view an image.

alert

Even if they have the passphrase option disabled it still prompts them.

My "I'm totally an artist" mockup:
yibfmub

But I'd recommend the box being **unchecked** by default and an option in settings to reenable the alert.

### PR DESCRIPTION
**Pull request by rhodey**
1) merge media preview and saving warnings into one dialog
2) remove old string resources

Fixes #1915
// FREEBIE

### CODE REVIEW

**Comment by rhodey** [comment]
@mcginty what do you think of the string resources? I'm not in love with them but tried to use your feedback from #2398

preview

**Comment by McLoo** [comment]
Never more settings/options, I know.
But what about re-enable warnings from advanced settings?

File: res/values/strings.xml [hunk 1]
```diff
@@ -170,6 +166,13 @@
     <string name="ExportFragment_error_while_writing_to_sd_card">Error while writing to SD card.</string>
     <string name="ExportFragment_success">Success!</string>
 
+    <!-- ExternalMediaWarningDialog -->
+    <string name="ExternalMediaWarningDialog_expose_secure_media">Expose secure media?</string>
+    <string name="ExternalMediaWarningDialog_previewing_or_saving_media_outside_of">Previewing or saving media outside of TextSecure allows other apps to see the media.</string>
```
Comment (mcginty) [inline res/values/strings.xml]: probably just "see it" instead of "see the media"

**Comment by rhodey** [comment]
@McLoo If we added this "don't ask again" option to more dialogs in the future I agree that an advanced "start asking again" (vocabulary to be determined :P) setting would be nice.

**Comment by rhodey** [comment]
@mcginty any further suggestions related to design or vocabulary? @moxie0 this diff can't get any smaller: new view, new strings, new preference, and swap out of dialog callbacks. in need of some feedback please.

File: src/org/thoughtcrime/securesms/MediaPreviewActivity.java [hunk 2]
```diff
@@ -224,9 +224,9 @@ protected void onPostExecute(Bitmap bitmap) {
   }
 
   private void saveToDisk() {
-    SaveAttachmentTask.showWarningDialog(this, new DialogInterface.OnClickListener() {
+    ExternalMediaWarningDialog.showIfNecessary(this, new WarningListener() {
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/MediaPreviewActivity.java]: why not just have the user pass in a ButtonCallback directly? Seems like the same amount of code for the caller and simplified code in the dialog class.

**Comment by mcginty** [comment]
I think for visual consistency the checkbox text should be styled the same as the main dialog content text.

**Comment by mcginty** [comment]
Also, to be consistent with material dialogs, the "CONTINUE" button should be "EXPOSE" in this case. Could consider using "reveal" instead of "expose".

**Comment by rhodey** [comment]
done, new screenshot screen

**Comment by rhodey** [comment]
the theme I was using to style the dialog text just got wiped out by 2d8de5291ef5b2867d38a33ac7afb2c4b4098b06, will reopen after switching to new text appearance style and testing

**Pull request closed without merging**